tests/functional/diff/__init__.py

```diff
@@ -0,0 +1,159 @@
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+"""Setup for the package tests."""
+
+import json
+import multiprocessing
+import os
+import shlex
+import shutil
+import subprocess
+import sys
+import time
+import uuid
+from subprocess import CalledProcessError
+
+from functional import TEST_ENV
+from functional import __REPO_ROOT
+
+from libtest import get_free_port
+from libtest import project
+from libtest import codechecker
+from libtest import env
+
+
+# Stopping event for CodeChecker server.
+__STOP_SERVER = multiprocessing.Event()
+TEST_WORKSPACE = None
+
+
+def setup_package():
+    """Setup the environment for the tests. Check the test project twice,
+    then start the server."""
+
+    tmp_dir = os.path.join(__REPO_ROOT, 'build')
+    base_dir = os.path.join(tmp_dir, 'workspace')
+
+    global TEST_WORKSPACE
+    TEST_WORKSPACE = env.get_workspace(base_dir, 'diff')
+
+    os.environ['TEST_WORKSPACE'] = TEST_WORKSPACE
+
+    test_project_path = project.path()
+
+    clang_version = env.clang_to_test()
+
+    pg_db_config = env.get_postgresql_cfg()
+
+    test_config = {}
+
+    project_info = project.get_info(project.path())
+
+    test_config['test_project'] = project_info
+
+    suppress_file = None
+
+    skip_list_file = None
+
+    # setup env vars for test cases
+    host_port_cfg = env.get_host_port_cfg()
+
+    codechecker_cfg = {
+        'suppress_file': suppress_file,
+        'skip_list_file': skip_list_file,
+        'check_env': TEST_ENV,
+        'use_postgresql': False,
+        'workspace': TEST_WORKSPACE,
+        'pg_db_config': pg_db_config,
+        'checkers': []
+    }
+
+    codechecker_cfg.update(host_port_cfg)
+
+    test_config['codechecker_cfg'] = codechecker_cfg
+
+    ret = project.clean(test_project_path, TEST_ENV)
+    if ret:
+        sys.exit(ret)
+
+    test_project_name_base = project_info['name'] + '_' + uuid.uuid4().hex
+
+    ret = codechecker.check(codechecker_cfg,
+                            test_project_name_base,
+                            test_project_path)
+    if ret:
+        sys.exit(1)
+    print("check was successful")
+    codechecker.wait_for_postgres_shutdown(TEST_WORKSPACE)
+    print("waiting postgresql to stop")
+
+    ret = project.clean(test_project_path, TEST_ENV)
+    if ret:
+        sys.exit(ret)
+
+    test_project_name_new = project_info['name'] + '_' + uuid.uuid4().hex
+
+    ret = codechecker.check(codechecker_cfg,
+                            test_project_name_new,
+                            test_project_path)
+    if ret:
+        sys.exit(1)
+    print("check was successful")
+    codechecker.wait_for_postgres_shutdown(TEST_WORKSPACE)
+    print("waiting postgresql to stop")
```

Comment:
These strings are inconsistent, some starts with a capital letter and some does not.